bson/__init__.py

```diff
@@ -1121,12 +1162,30 @@ def _decode_selective(rawdoc: Any, fields: Any, codec_options: Any) -> Mapping[A
 
 def _convert_raw_document_lists_to_streams(document: Any) -> None:
     cursor = document.get("cursor")
-    if cursor:
-        for key in ("firstBatch", "nextBatch"):
-            batch = cursor.get(key)
-            if batch:
-                stream = b"".join(doc.raw for doc in batch)
-                cursor[key] = [stream]
+    if not cursor:
+        return
+    for key in ("firstBatch", "nextBatch"):
+        batch = cursor.get(key)
+        if not batch:
+            continue
+
+        # Extract the raw bytes of each document.
+        data, _ = get_data_and_view(batch.raw)
```

Comment:
Should this be:
```
        _, data = get_data_and_view(batch.raw)
```
That way we use a memory view to avoid making many small byte copies.

Reply:
Done

Reply:
It turns out we needed the bytes so we can use the `index` method.

Reply:
Okay, back to using memoryview directly.